codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketServerProtocolHandler.java

```diff
@@ -57,10 +58,32 @@
     public enum ServerHandshakeStateEvent {
         /**
          * The Handshake was complete succesful and so the channel was upgraded to websockets
+         *
+         * @deprecated in favor of {@link HandshakeComplete} class,
+         * it provides extra information about the handshake
          */
+        @Deprecated
         HANDSHAKE_COMPLETE
     }
 
+    /**
+     * Marker interface for events that are fired to notify about handshake status
+     */
+    public interface ServerHandshakeEvent {  }
+
+    /**
+     * The Handshake was complete succesful and so the channel was upgraded to websockets
+     */
+    public static class HandshakeComplete implements ServerHandshakeEvent {
+        public final FullHttpRequest upgradeRequest;
```

Comment:
Wasn't aware of the counting. I'll keep just a copy of the URI and the headers. I don't need it in my use case, but I think it could be useful sometimes for WS channel initialization to have easy access to those request attributes.